client/api_provider.go

```diff
@@ -114,16 +112,36 @@ type apiProvider struct {
 	rk          *RepositoryKey
 }
 
-// Should only be called on initialization.
 // This performs an exponential backoff until the context is cancelled.
-func (a *apiProvider) register(ctx context.Context) error {
+// We do this to try to alleviate some issues with sidecars starting up.
+func (a *apiProvider) registerWithBackoff(ctx context.Context) error {
 	req := connect.NewRequest(&clientv1beta1.RegisterRequest{RepoKey: a.rk.toProto()})
 	req.Header().Set(LekkoAPIKeyHeader, a.apikey)
+	ticker := backoff.NewTicker(backoff.NewExponentialBackOff())
 	op := func() error {
 		_, err := a.lekkoClient.Register(ctx, req)
 		return err
 	}
-	return backoff.Retry(op, backoff.WithContext(backoff.NewExponentialBackOff(), ctx))
+	for {
+		var err error
+		if err = op(); err == nil {
+			return nil
+		}
+		// else, we have an error, but we keep it around to retry.
+		select {
+		case <-ctx.Done():
+			return errors.Wrap(err, "timed out retrying, returning last error")
```

Comment:
should we wrap all the errors leading up to and including the last error instead of just the last error?